dependencies.yaml

```diff
@@ -0,0 +1,107 @@
+# Dependency list for https://github.com/rapidsai/dependency-file-generator
+ files:
+   all:
+     output: conda
+     matrix:
+       cuda: ["11.5"]
+       arch: [x86_64]
+     includes:
+       - common_build
+       - cpp_build
+       - cudatoolkit
+       - doc
+       - python_build
+ channels:
+   - rapidsai
+   - rapidsai-nightly
+   - dask/label/dev
+   - conda-forge
+   - nvidia
+ dependencies:
+   cudatoolkit:
+     specific:
+       - output_types: conda
+         matrices:
+           - matrix:
+               cuda: "11.5"
+             packages:
+               - cudatoolkit=11.5
+           - matrix:
+               cuda: "11.4"
+             packages:
+               - cudatoolkit=11.4
+           - matrix:
+               cuda: "11.2"
+             packages:
+               - cudatoolkit=11.2
+           - matrix:
+               cuda: "11.0"
+             packages:
+               - cudatoolkit=11.0
+   common_build:
+     common:
+       - output_types: [conda]
+         packages:
+           - cmake>=3.23.1,!=3.25.0
+       - output_types: conda
+         packages:
+           - ninja
+           - pip
+   cpp_build:
+     common:
+       - output_types: conda
+         packages:
+           - nvcc_linux-64=11.5
+           - c-compiler
+           - cxx-compiler
+           - librmm=23.02.*
+           - libraft-headers=23.02.*
+           - libraft-distance=23.02.*
+           - libcugraphops=23.02.*
+           - clang=11.1.0
+           - clang-tools=11.1.0
```

Comment:
I think these can be dropped unless you know a reason why not. Historically these were used only to provide clang-format, as I understand it.

Reply:
I'm OK with removing things.  If we remove these from here, how does the C++ code formatting work?  Is there some other mechanism we are using?

Reply:
Many other RAPIDS repos have migrated to using `pre-commit` as a single source of truth for CI style checks. I will file a PR to update cugraph to match, since that's something we've been doing as part of the GitHub Actions migrations.

For clang-format specifically, switching to pre-commit lets us eliminate this section:
https://github.com/rapidsai/cugraph/blob/efcf6a8e9bd07936fd371d16eb2563502c383ca3/ci/checks/style.sh#L33-L36
as well as the entire `run-clang-format.py` file in favor of a single hook definition [like this example from cudf](https://github.com/rapidsai/cudf/blob/0c8eeb5a3caacf5881621f84da891a5d804d43d7/.pre-commit-config.yaml#L62-L67).

Then the style check script can also be simplified from [~79 lines](https://github.com/rapidsai/cugraph/blob/efcf6a8e9bd07936fd371d16eb2563502c383ca3/ci/checks/style.sh) to [~23 lines, like cudf](https://github.com/rapidsai/cudf/blob/branch-23.02/ci/check_style.sh).